crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -127,4 +129,53 @@ impl Database for MongoDb {
             .await
             .expect("Failed to fetch latest job by given job type"))
     }
+
+    async fn get_successful_snos_jobs_without_proving(&self) -> Result<Vec<JobItem>> {
+        let filter = vec![
+            // Stage 1: Match successful SNOS job runs
+            doc! {
+                "$match": {
+                    "job_type": "SnosRun",
+                    "status": "Completed",
+                }
+            },
+            // Stage 2: Lookup to find corresponding proving jobs
+            doc! {
+                "$lookup": {
+                    "from": "jobs",
+                    "let": { "internal_id": "$internal_id" },
+                    "pipeline": [
+                        {
+                            "$match": {
+                                "$expr": {
+                                    "$and": [
+                                        { "$eq": ["$job_type", "ProofCreation"] },
+                                        { "$eq": ["$internal_id", "$$internal_id"] }
```

Comment:
Hey @unstark 

So I talked about this to apoorv and he told me that the interal_id field was just to make sure that for a particular job type there is a way to track the jobs internally so that's why a generic name `internal_id` was given to this field.

Reply:
So @unstark my hypothesis was that the orchestrator can be designed in a way that we can also have jobs at the txn level later on. Maybe something like txn level proving or stream of txs to some XYZ place. So in that case, the internal_id will be the txn_hash. What are your thoughts on this? Does it feel like an overkill?